jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/PropertyRule.java

```diff
@@ -129,6 +129,24 @@ private boolean isRequired(String nodeName, JsonNode node, Schema schema) {
         return false;
     }
 
+    private boolean usesOptional(String nodeName, JsonNode node, Schema schema) {
+        if (node.has("java_optional")) {
+            final JsonNode requiredNode = node.get("java_optional");
+            return requiredNode.asBoolean();
+        }
+
+        JsonNode requiredArray = schema.getContent().get("java_optional");
```

Comment:
I think you mean `javaOptionalArray` here? `requiredArray` is a name which I assume was taken from another method.